src/odemis/acq/stream/_sync.py

```diff
@@ -136,13 +136,22 @@ def __init__(self, name, streams):
                 break
 
         # Get polarization analyzer if found in optical substream
+        self._analyzer = None
         for s in streams:
             # check if stream has attribute analyzer and analyzer is not None
             if hasattr(s, "analyzer") and s.analyzer:
-                # get polarization analyzer
-                self._analyzer = s.analyzer
-                # get polarization analyzer VA
-                self._polarization = s.polarization
+                # additionally check that not multiple streams have a component analyzer
+                if not self._analyzer:
```

Comment:
A few notes:
* When handling such error, it's nice to not put the standard code inside the if, so that it avoids having an extra sub-level in the code and it makes it easier to read.
* Never raise directly an `Exception` base class, use a sub-class. Here, `ValueError` is fine.
* Try to keep the exception message short (~one line).

So I'd suggest it write it this way:
```
if hasattr(s, "analyzer") and s.analyzer:
    if self._analyzer:
         raise ValueError("Only one stream can have an analyzer specified")
    # get polarization analyzer and the VA with the requested position(s)
    self._analyzer = s.analyzer
    self._polarization = s.polarization
```